src/Services/MediaService.php

```diff
@@ -51,9 +48,12 @@ public function create($content, string $fileName, array $data = []): Model
         $data['name'] = $fileName;
         $data['link'] = Storage::url($data['name']);
         $data['owner_id'] = Auth::id();
-        $data['preview_id'] = $this->createPreview($data['name'])->id;
+        $preview = $this->createPreview($data['name']);
+        $data['preview_id'] = $preview->id;

```

Comment:
please move preview generation before data array modification

```suggestion
        $preview = $this->createPreview($data['name']);

        $data['preview_id'] = $preview->id;
        $data['name'] = $fileName;
        // .... other fields setting
```